tools/istio-docker.mk

```diff
@@ -19,10 +19,10 @@
 
 # Docker target will build the go binaries and package the docker for local testing.
 # It does not upload to a registry.
-docker: build-linux docker.all
+docker: docker.all
```

Comment:
targets already do a build

Reply:
But on a Mac, the build will generate only the Darwin binaries.

Reply:
I am fairly sure the rules like `docker.citadel: $(ISTIO_DOCKER)/istio_ca`  are doing the right thing and this is completely redundant. Otherwise rules like `make docker.citadel` would fail. Would love if you could verify though

Reply:
But I see that the _docker._ targets list the `$(ISTIO_OUT_LINUX)...` targets as dependencies so they will get built as part of the _docker_ target. So this is fine.